src/ToolingSharedLibrary/Includes/Utils/Helpers.h

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#pragma once
+#include <pch.h>
+#include <span>
+#include <ranges>
+
+namespace Helpers
+{
+    // Simple OrderedMap class to maintain insertion order and fast lookup
+    template<typename Key, typename Value>
+    class OrderedMap 
+    {
+        public:
+            Value& at(const Key& key) { return m_map.at(key); }
+            const Value& at(const Key& key) const { return m_map.at(key); }
+            auto find(const Key& key) { return m_map.find(key); }
+            auto find(const Key& key) const { return m_map.find(key); }
+            std::span<const Key> keys() const { return m_keys; }
+            size_t size() const { return m_map.size(); }
+
+            // Technically if this was a true orderedmap doing begin() should return [key in order, value in order]
+            // not the unordered_map version. However we don't need this for our use cases. We just need this classes 
+            // keys() and values() methods to return their data in insertion order.
+            auto begin() const { return m_map.begin(); }
+            auto end() const { return m_map.end(); }
+            auto begin() { return m_map.begin(); }
+            auto end() { return m_map.end(); }
+
+            const Value& operator[](const Key& key) const { return m_map.at(key); }
+            bool contains(const Key& key) const { return m_map.contains(key); }
+            bool empty() const { return m_map.empty(); }
+            void clear()
+            {
+                m_map.clear();
+                m_keys.clear();
+            }
+
+            Value& operator[](const Key& key)
+            {
+                if (!contains(key))
+                {
+                    m_keys.push_back(key);
+                }
+
+                return m_map[key];
+            }
+
+
+            void insert(const Key key, const Value& value)
+            {
+                if (!contains(key))
+                {
+                    m_keys.push_back(key);
+                }
+
+                m_map[key] = value;
+            }
+
```

Comment:
nit: i think you have 2 extra copies.

            void insert(const Key key, Value&& value)
            {
                if (!contains(key))
                {
                    m_keys.push_back(key);
                }

                m_map.insert(std::move(value));
            }

Reply:
wont fix is fine